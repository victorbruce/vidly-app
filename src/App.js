import React from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
import './App.css';
import Navigation from './components/navigation';
import Movies from './components/movies';
import MovieForm from './components/movieForm';
import Customers from './components/customers';
import Rentals from './components/rentals';
import NotFound from './components/not-found';

function App() {
  return (
    <React.Fragment>
      <Navigation />
      <main className="container">
        <Switch>
          <Route path="/movies/:id" component={MovieForm} />
          <Route path="/movies" component={Movies} />
          <Route path="/rentals" component={Rentals} />
          <Route path="/customers" component={Customers} />
          <Route path="/not-found" component={NotFound} />
          <Redirect from="/" exact to="/movies" />
          <Redirect to="/not-found" />
        </Switch>
      </main>
    </React.Fragment>
  );
}

export default App;
