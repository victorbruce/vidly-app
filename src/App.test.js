import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import App from './App';
import renderer from 'react-test-renderer';

test('snapshot App', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
