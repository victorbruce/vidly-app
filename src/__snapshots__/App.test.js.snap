// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot App 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light rounded nav-bottom"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Vidly
    </a>
    <button
      aria-controls="navbarsExample09"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarsExample09"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarsExample09"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/movies"
            onClick={[Function]}
            style={Object {}}
          >
            Movies 
            <span
              className="sr-only"
            >
              (current)
            </span>
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/customers"
            onClick={[Function]}
          >
            Customers
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/rentals"
            onClick={[Function]}
          >
            Rentals
          </a>
        </li>
      </ul>
    </div>
  </nav>,
  <main
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-3"
      >
        <ul
          className="list-group"
        >
          <li
            className="list-group-item"
            onClick={[Function]}
          >
            All Genres
          </li>
          <li
            className="list-group-item"
            onClick={[Function]}
          >
            Action
          </li>
          <li
            className="list-group-item"
            onClick={[Function]}
          >
            Comedy
          </li>
          <li
            className="list-group-item"
            onClick={[Function]}
          >
            Thriller
          </li>
        </ul>
      </div>
      <div
        className="col"
      >
        <p>
          Showing 
          9
           movies in the database.
        </p>
        <table
          className="table table-striped"
        >
          <thead>
            <tr>
              <th
                className="clickable"
                onClick={[Function]}
              >
                Title
                 
              </th>
              <th
                className="clickable"
                onClick={[Function]}
              >
                Genre
                 
              </th>
              <th
                className="clickable"
                onClick={[Function]}
              >
                Stock
                 
              </th>
              <th
                className="clickable"
                onClick={[Function]}
              >
                Rate
                 
              </th>
              <th
                className="clickable"
                onClick={[Function]}
              >
                 
              </th>
              <th
                className="clickable"
                onClick={[Function]}
              >
                 
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <a
                  href="/movies/5b21ca3eeb7f6fbccd47181a"
                  onClick={[Function]}
                >
                  Airplane
                </a>
              </td>
              <td>
                Comedy
              </td>
              <td>
                7
              </td>
              <td>
                3.5
              </td>
              <td>
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-heart fa-w-16 "
                  data-icon="heart"
                  data-prefix="far"
                  focusable="false"
                  onClick={[Function]}
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </td>
              <td>
                <button
                  className="btn btn-danger btn-sm"
                  onClick={[Function]}
                >
                  Delete
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <a
                  href="/movies/5b21ca3eeb7f6fbccd471816"
                  onClick={[Function]}
                >
                  Die Hard
                </a>
              </td>
              <td>
                Action
              </td>
              <td>
                5
              </td>
              <td>
                2.5
              </td>
              <td>
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-heart fa-w-16 "
                  data-icon="heart"
                  data-prefix="far"
                  focusable="false"
                  onClick={[Function]}
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </td>
              <td>
                <button
                  className="btn btn-danger btn-sm"
                  onClick={[Function]}
                >
                  Delete
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <a
                  href="/movies/5b21ca3eeb7f6fbccd471817"
                  onClick={[Function]}
                >
                  Get Out
                </a>
              </td>
              <td>
                Thriller
              </td>
              <td>
                8
              </td>
              <td>
                3.5
              </td>
              <td>
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-heart fa-w-16 "
                  data-icon="heart"
                  data-prefix="far"
                  focusable="false"
                  onClick={[Function]}
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </td>
              <td>
                <button
                  className="btn btn-danger btn-sm"
                  onClick={[Function]}
                >
                  Delete
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <a
                  href="/movies/5b21ca3eeb7f6fbccd47181e"
                  onClick={[Function]}
                >
                  Gone Girl
                </a>
              </td>
              <td>
                Thriller
              </td>
              <td>
                7
              </td>
              <td>
                4.5
              </td>
              <td>
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-heart fa-w-16 "
                  data-icon="heart"
                  data-prefix="far"
                  focusable="false"
                  onClick={[Function]}
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </td>
              <td>
                <button
                  className="btn btn-danger btn-sm"
                  onClick={[Function]}
                >
                  Delete
                </button>
              </td>
            </tr>
          </tbody>
        </table>
        <div>
          <nav
            aria-label="Page navigation example"
          >
            <ul
              className="pagination"
            >
              <li
                className="page-item"
              >
                <a
                  className="page-link"
                  href="#"
                >
                  Previous
                </a>
              </li>
              <li
                className="page-item active"
              >
                <a
                  className="page-link"
                  href="#"
                  onClick={[Function]}
                >
                  1
                </a>
              </li>
              <li
                className="page-item"
              >
                <a
                  className="page-link"
                  href="#"
                  onClick={[Function]}
                >
                  2
                </a>
              </li>
              <li
                className="page-item"
              >
                <a
                  className="page-link"
                  href="#"
                  onClick={[Function]}
                >
                  3
                </a>
              </li>
              <li
                className="page-item"
              >
                <a
                  className="page-link"
                  href="#"
                >
                  Next
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </main>,
]
`;

exports[`snapshot test 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light rounded nav-bottom"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Vidly
    </a>
    <button
      aria-controls="navbarsExample09"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarsExample09"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarsExample09"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/movies"
            onClick={[Function]}
            style={Object {}}
          >
            Movies 
            <span
              className="sr-only"
            >
              (current)
            </span>
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/customers"
            onClick={[Function]}
          >
            Customers
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/rentals"
            onClick={[Function]}
          >
            Rentals
          </a>
        </li>
      </ul>
    </div>
  </nav>,
  <main
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-3"
      >
        <ul
          className="list-group"
        >
          <li
            className="list-group-item"
            onClick={[Function]}
          >
            All Genres
          </li>
          <li
            className="list-group-item"
            onClick={[Function]}
          >
            Action
          </li>
          <li
            className="list-group-item"
            onClick={[Function]}
          >
            Comedy
          </li>
          <li
            className="list-group-item"
            onClick={[Function]}
          >
            Thriller
          </li>
        </ul>
      </div>
      <div
        className="col"
      >
        <p>
          Showing 
          9
           movies in the database.
        </p>
        <table
          className="table table-striped"
        >
          <thead>
            <tr>
              <th
                className="clickable"
                onClick={[Function]}
              >
                Title
                 
              </th>
              <th
                className="clickable"
                onClick={[Function]}
              >
                Genre
                 
              </th>
              <th
                className="clickable"
                onClick={[Function]}
              >
                Stock
                 
              </th>
              <th
                className="clickable"
                onClick={[Function]}
              >
                Rate
                 
              </th>
              <th
                className="clickable"
                onClick={[Function]}
              >
                 
              </th>
              <th
                className="clickable"
                onClick={[Function]}
              >
                 
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <a
                  href="/movies/5b21ca3eeb7f6fbccd47181a"
                  onClick={[Function]}
                >
                  Airplane
                </a>
              </td>
              <td>
                Comedy
              </td>
              <td>
                7
              </td>
              <td>
                3.5
              </td>
              <td>
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-heart fa-w-16 "
                  data-icon="heart"
                  data-prefix="far"
                  focusable="false"
                  onClick={[Function]}
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </td>
              <td>
                <button
                  className="btn btn-danger btn-sm"
                  onClick={[Function]}
                >
                  Delete
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <a
                  href="/movies/5b21ca3eeb7f6fbccd471816"
                  onClick={[Function]}
                >
                  Die Hard
                </a>
              </td>
              <td>
                Action
              </td>
              <td>
                5
              </td>
              <td>
                2.5
              </td>
              <td>
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-heart fa-w-16 "
                  data-icon="heart"
                  data-prefix="far"
                  focusable="false"
                  onClick={[Function]}
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </td>
              <td>
                <button
                  className="btn btn-danger btn-sm"
                  onClick={[Function]}
                >
                  Delete
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <a
                  href="/movies/5b21ca3eeb7f6fbccd471817"
                  onClick={[Function]}
                >
                  Get Out
                </a>
              </td>
              <td>
                Thriller
              </td>
              <td>
                8
              </td>
              <td>
                3.5
              </td>
              <td>
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-heart fa-w-16 "
                  data-icon="heart"
                  data-prefix="far"
                  focusable="false"
                  onClick={[Function]}
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </td>
              <td>
                <button
                  className="btn btn-danger btn-sm"
                  onClick={[Function]}
                >
                  Delete
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <a
                  href="/movies/5b21ca3eeb7f6fbccd47181e"
                  onClick={[Function]}
                >
                  Gone Girl
                </a>
              </td>
              <td>
                Thriller
              </td>
              <td>
                7
              </td>
              <td>
                4.5
              </td>
              <td>
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-heart fa-w-16 "
                  data-icon="heart"
                  data-prefix="far"
                  focusable="false"
                  onClick={[Function]}
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </td>
              <td>
                <button
                  className="btn btn-danger btn-sm"
                  onClick={[Function]}
                >
                  Delete
                </button>
              </td>
            </tr>
          </tbody>
        </table>
        <div>
          <nav
            aria-label="Page navigation example"
          >
            <ul
              className="pagination"
            >
              <li
                className="page-item"
              >
                <a
                  className="page-link"
                  href="#"
                >
                  Previous
                </a>
              </li>
              <li
                className="page-item active"
              >
                <a
                  className="page-link"
                  href="#"
                  onClick={[Function]}
                >
                  1
                </a>
              </li>
              <li
                className="page-item"
              >
                <a
                  className="page-link"
                  href="#"
                  onClick={[Function]}
                >
                  2
                </a>
              </li>
              <li
                className="page-item"
              >
                <a
                  className="page-link"
                  href="#"
                  onClick={[Function]}
                >
                  3
                </a>
              </li>
              <li
                className="page-item"
              >
                <a
                  className="page-link"
                  href="#"
                >
                  Next
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </main>,
]
`;
